ChronoVisor/include/ClientInfo.h

```diff
@@ -5,14 +5,47 @@
 #ifndef CHRONOLOG_CLIENTINFO_H
 #define CHRONOLOG_CLIENTINFO_H
 
+#include <sstream>
+
+class Story;
 
 class ClientInfo {
 public:
     ClientInfo() = default;
 
+    friend std::ostream &operator<<(std::ostream &out, const ClientInfo &r) {
+        std::stringstream ss;
+        auto it = r.acquiredStoryList_.begin();
+        ss << it->first;
+        for (++it; it != r.acquiredStoryList_.end(); ++it) {
+            ss << ", " << it->first;
+        }
+        return out << r.addr_ << ":" << r.port_ << ", acquiredStories: " << ss.str();
+    }
+
+    [[nodiscard]] std::string to_string() const {
+        std::string str = addr_ + ":" + std::to_string(port_) + ", acquiredStories: ";
+        std::stringstream ss;
+        auto it = acquiredStoryList_.begin();
+        ss << it->first;
+        for (++it; it != acquiredStoryList_.end(); ++it) {
+            ss << ", " << it->first;
+        }
+        return str + ss.str();
+    }
+
+    template<typename A> friend void serialize(A& ar, ClientInfo& r);
+
     std::string addr_;
-    uint16_t port_{};
+    uint16_t port_;
+    std::unordered_map<uint64_t, Story *> acquiredStoryList_;
 };
 
```

Comment:
why do we need to serialize ClientInfo ? It's an internal structure for ClientRegistry, we don't put it on the wire, do we?

Reply:
This is a history problem. We had the design of writing ClientRegistry periodically to disk to store the history statistics of client behaviors for things like custom configuration and optimization. If we do not have that design anymore, I can delete it. But I think it is also okay to leave it there.